### PR TITLE
Issue#240: Moved home-server-generator module into home-application.

### PR DESCRIPTION
**Pull request by Illia-Shchur**
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by kh0ma — CHANGES_REQUESTED** [review]
Please look at the issue description. The idea was to inline home-server-generator into home-application module.

File: home-application/pom.xml [hunk 1]
```diff
@@ -88,6 +88,34 @@
...
+                <executions>
+                    <execution>
+                        <id>resource-dependencies</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
```
Comment (kh0ma) [inline home-application/pom.xml]: Why do we need to do it twice? I see that we've already been unpacking it here below in the pom. Just change the phase if needed.

File: home-application/pom.xml [hunk 2]
```diff
@@ -78,13 +80,30 @@
     <build>
         <plugins>
             <plugin>
-                <groupId>org.springframework.boot</groupId>
-                <artifactId>spring-boot-maven-plugin</artifactId>
```
Comment (kh0ma) [inline home-application/pom.xml:82]: Why did you remove this plugin? In order to run the app from jar we still need fat jar.

File: home-application/pom.xml [hunk 3]
```diff
@@ -78,13 +80,30 @@
...
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>${maven.dependency.plugin.version}</version>
```
Comment (kh0ma) [inline home-application/pom.xml:83]: Please do not add any additional `maven-dependency-plugin` execution. Just rely on the existing.

**Review by kh0ma — CHANGES_REQUESTED** [review]

File: home-application/pom.xml (hunk 2 above)
Comment (Illia-Shchur) [reply home-application/pom.xml:82]: that was not deliberately, will return plugin in the next commit

File: home-application/pom.xml (hunk 3 above)
Comment (Illia-Shchur) [reply home-application/pom.xml:85]: i didn't add any additional plugin, just changed their order

File: home-application/pom.xml (hunk 1 above)
Comment (Illia-Shchur) [reply home-application/pom.xml]: this one will beremoved in the next commit.

File: home-application/pom.xml (hunk 3 above)
Comment (Illia-Shchur) [reply home-application/pom.xml:83]: this dependency was taken from home-server-generator. It will be removed in the next commit

**Review by kizimov — APPROVED** [review]
LGTM

**Review by likeRewca — APPROVED** [review]

File: home-application/pom.xml [hunk 4]
```diff
@@ -182,6 +206,17 @@
...
+                        <goals>
+                            <goal>repackage</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
         </plugins>
     </build>
-</project>
+</project>
```
Comment (kh0ma) [inline home-application/pom.xml]: Please do not forget to add nextline at the end of the file.

**Review by kh0ma — APPROVED** [review]

**Review by kizimov — APPROVED** [review]

**Review by AriannaLiss — APPROVED** [review]
